server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -738,9 +738,9 @@ protected Node(final Environment initialEnvironment,
     }
 
     // package scope for testing
-    List<NamedXContentRegistry.Entry> getCompatibleNamedXContents() {
+    Stream<NamedXContentRegistry.Entry> getCompatibleNamedXContents() {
```

Comment:
I'm a bit suspicious about this switch from List to Stream -- can you walk it back or convince me that it's a great idea?

Reply:
the reason is that when `NamedXContentRegistry` is created here https://github.com/elastic/elasticsearch/pull/75722/files#diff-d01b960cab84fe2338f230b5e7535bc8efa9f9778ecf61c9f527067351dca895R413
it aggregates all streams before converting to a list. I was thinking this could avoid unnecessary list creation in `getCompatibleNamedXContents`
but.. It made me thinking if we even need this now. We can register both compatible and current NamedXContentEntries in the same registry..
I guess I should just remove that `Node#getCompatibleNamedXContents`  and `Plugin#getNamedXContentForCompatibility`

Reply:
This is how usage of shared registry for compatible and current `NamedXContentRegistry.Entry` entries will look like
draft for type query:
https://github.com/elastic/elasticsearch/pull/75453/files#diff-57f10f7614a75787774b36d41833688a0474b84eb239e4c84499feec9634eefbR845